src/filtering/smooth.jl

```diff
@@ -103,20 +103,52 @@ function smooth!(
 
     # x_curr.μ .+= G * (x_next.μ .- x_pred.μ) # less allocations:
     x_pred.μ .-= x_next.μ
-    _matmul!(x_curr.μ, G, x_pred.μ, -1, 1)
+    a = D ÷ _D
+    _matmul!(reshape_no_alloc(x_curr.μ, _D, a), G, reshape_no_alloc(x_pred.μ, _D, a), -1, 1)
 
     # Joseph-Form:
     R = C_3DxD
 
     G2 = _matmul!(C_DxD, G, Ah)
-    copy!(view(R, 1:D, 1:D), x_curr.Σ.R)
-    _matmul!(view(R, 1:D, 1:D), x_curr.Σ.R, G2', -1.0, 1.0)
+    copy!(view(R, 1:_D, 1:_D), x_curr.Σ.R)
+    _matmul!(view(R, 1:_D, 1:_D), x_curr.Σ.R, G2', -1.0, 1.0)
 
-    _matmul!(view(R, D+1:2D, 1:D), Qh.R, _matmul!(G2, G, sqrt.(diffusion))')
-    _matmul!(view(R, 2D+1:3D, 1:D), x_next.Σ.R, G')
+    _matmul!(view(R, _D+1:2_D, 1:_D), Qh.R, _matmul!(G2, G, sqrt.(diffusion))')
+    _matmul!(view(R, 2_D+1:3_D, 1:_D), x_next.Σ.R, G')
 
     Q_R = triangularize!(R, cachemat=C_DxD)
     copy!(x_curr.Σ.R, Q_R)
 
     return nothing
 end
+
+
+function smooth!(
+    x_curr::SRGaussian{T,<:Kronecker.KroneckerProduct},
```

Comment:
As before: Make sure the type check is the correct one and check that the left factors are all I